File: Python/day2/AOC.py
def parse_input(input):
    dims = []
    for line in input:
        l, w, h = list(map(int, line.split('x')))
        dims.append([l, w, h])
    return dims

def part1(dims):
    total = 0
    for dim in dims:
        l, w, h = dim
        s1 = l * w
        s2 = w * h
        s3 = h * l
        small = min(s1, s2, s3)
        total += 2 * (s1 + s2 + s3) + small
    return total

def part2(dims):
    total = 0
    for dim in dims:
        l, w, h = dim
        r1, r2, _ = sorted(dim)
        total += 2 * r1 + 2 * r2 + l * w * h
    return total

def main():
    input = open("input.txt", "r").read().splitlines()
    part1_sum = part1(parse_input(input))
    part2_sum = part2(parse_input(input))
    print(f"Part 1: {part1_sum}")
    print(f"Part 2: {part2_sum}")

if __name__ == "__main__":
    main()

File: Go/day02/AOC.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func part1(input []string) int {
	total := 0
	for _, line := range input {
		var dims []int
		for _, dim := range strings.Split(line, "x") {
			d, _ := strconv.Atoi(dim)
			dims = append(dims, d)
		}

		l, w, h := dims[0], dims[1], dims[2]
		sides := []int{l * w, w * h, h * l}
		sort.Ints(sides)
		for _, side := range sides {
			total += 2 * side
		}

		total += sides[0]
	}
	return total
}

func part2(input []string) int {
	total := 0
	for _, line := range input {
		var dims []int
		for _, dim := range strings.Split(line, "x") {
			d, _ := strconv.Atoi(dim)
			dims = append(dims, d)
		}

		sort.Ints(dims)
		r1, r2 := dims[0], dims[1]

		total += 2*r1 + 2*r2 + r1*r2*dims[2]
	}
	return total
}

func main() {
	input, err := os.ReadFile("input.txt")
	if err != nil {
		fmt.Print("Error reading file\n")
		panic(err)
	}

	lines := strings.Split(string(input), "\n")
	part1_ans := part1(lines)
	fmt.Printf("Part 1: %d\n", part1_ans)

	part2_ans := part2(lines)
	fmt.Printf("Part 2: %d\n", part2_ans)
}
